In [1]:
import pandas as pd

aggregate functions = Reduces a set of values into a single summary value.
                      Used to summarize and analyze data.
                      Often used with the groupby() function

In [2]:
#Basically we are reducing a bunch of values into a single one
df = pd.read_csv("pokemon.csv")
print(df)

       #                   Name   Type 1  Type 2  Total  HP  Attack  Defense  \
0      1              Bulbasaur    Grass  Poison    318  45      49       49   
1      2                Ivysaur    Grass  Poison    405  60      62       63   
2      3               Venusaur    Grass  Poison    525  80      82       83   
3      3  VenusaurMega Venusaur    Grass  Poison    625  80     100      123   
4      4             Charmander     Fire     NaN    309  39      52       43   
..   ...                    ...      ...     ...    ...  ..     ...      ...   
795  719                Diancie     Rock   Fairy    600  50     100      150   
796  719    DiancieMega Diancie     Rock   Fairy    700  50     160      110   
797  720    HoopaHoopa Confined  Psychic   Ghost    600  80     110       60   
798  720     HoopaHoopa Unbound  Psychic    Dark    680  80     160       60   
799  721              Volcanion     Fire   Water    600  80     110      120   

     Sp. Atk  Sp. Def  Speed  Generatio

We have a dataframe that contains lists of all pokemon from Gen 1 kanto to Gen 6 kalos.

Now lets say we want mean of all values from a column

    Default syntax: print(df.mean())

    But this sends error because in our dataframe their contains columns that have string and we can't calculate mean of string.

So we use print(df.mean(numeric_only = True))

    We pass in the kyword argument in df.mean() called 'numeric_only'.
    We are telling pandas to find the mean of those columns that are numeric.


In [3]:
print(df.mean(numeric_only = True))

#             362.81375
Total         435.10250
HP             69.25875
Attack         79.00125
Defense        73.84250
Sp. Atk        72.82000
Sp. Def        71.90250
Speed          68.27750
Generation      3.32375
Legendary       0.08125
dtype: float64


Now what does these data represent?

From gen 1 to gen 2, The average total stat of all pokemons is 435.10.
This means most are way below the standard strong stats of 600 for pseudo legendaries and 680 for box art legendaries.

Most also have really bad atk stats with average only being 79.001. Keep in mind pokemon like Mega Rayqueza have atk stats of 150 {Which is super high}

Similar is the case for sp.atk stats as pokemon like mega mewtwo Y have sp atk stat of 194 which is highest for all legendaries.

In [7]:
#                   Another aggregate function is the sum()
print(df.sum(numeric_only=True))
#                   This sums all the numeric values of each pokemon in a column

#             290251
Total         348082
HP             55407
Attack         63201
Defense        59074
Sp. Atk        58256
Sp. Def        57522
Speed          54622
Generation      2659
Legendary         65
dtype: int64


Most of the sum data from above is useless Except one which is:

        Their are 65 legendary from pokemon generation 1 to generation 6.